avro-codegen/src/main/java/com/linkedin/avroutil1/codegen/SpecificRecordClassGenerator.java

```diff
@@ -73,4 +74,46 @@ protected JavaFileObject generateSpecificEnum(AvroEnumSchema enumSchema, Specifi
 
     return javaFile.toJavaFileObject();
   }
+
+  /**
+   * adds "public final static Schema SCHEMA$" field to generated classes for named avro types
+   * @param classBuilder
+   * @param classSchema
+   */
+  protected void addSchema$ToGeneratedClass(TypeSpec.Builder classBuilder, AvroNamedSchema classSchema) {
+    ClassName avroSchemaType = ClassName.get("org.apache.avro", "Schema");
+    classBuilder.alwaysQualify(avroSchemaType.simpleName()); //no import statements
+
+    //get fully-inlined single-line avsc from schema
+    AvscSchemaWriter avscWriter = new AvscSchemaWriter();
+    String avsc = avscWriter.writeSingle(classSchema).getContents();
+
+    //JVM spec spec says string literals cant be over 65535 bytes in size (this isnt simply the
+    //character count as horrible wide unicode characters could be involved).
+    //for details see https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-4.html#jvms-4.4.7
+    //we add some extra safety margin
+    String parseFormat;
+    Object[] parseFormatArgs;
+    if (avsc.getBytes(StandardCharsets.UTF_8).length > 64000) {
+      //not 100% safe as argument is in characters and should be bytes ...
+      List<String> chunks = SourceCodeUtils.safeSplit(avsc, 20000);
+      StringJoiner csv = new StringJoiner(", ");
+      for (int i = 1; i <= chunks.size(); i++) {
+        //"$1S, $2S, ... $NS"
+        csv.add("$" + i + "S");
+      }
+      parseFormat = HelperConsts.HELPER_FQCN + ".parse(" + csv + ")";
+      parseFormatArgs = chunks.toArray(new Object[] {});
+    } else {
+      //no need to split anything
+      parseFormat = HelperConsts.HELPER_FQCN + ".parse($1S)";
+      parseFormatArgs = new Object[] {avsc};
+    }
+    classBuilder.addField(FieldSpec
+            .builder(avroSchemaType, "SCHEMA$", Modifier.PUBLIC, Modifier.FINAL, Modifier.STATIC)
+            //TODO - use strict parsing
+            .initializer(CodeBlock.of(parseFormat, parseFormatArgs))
+            .build()
```

Comment:
It might be useful to describe what the final generated code will look like: 

```
public final static Schema SCHEMA$ = com.linkedin.avroutil1.compatibility.AvroCompatibilityHelper.parse(...
```

Reply:
added to javadoc for method